### ISSUE TITLE
Move type assertion to lib

### ISSUE THREAD

**Issue by nbrl** [digitalanimal/advocate-things-js-sdk#57]
We should move all of our type assertions to a library to keep the main SDK cleaner.

@jablab wrote in PR #55:

> Reckon it might be worth creating a new file in lib/ for type checking functions? e.g.

> ```js
// lib/typeChecks.js
this.isObject = function (x) {
    return Object.prototype.toString.call(x) === '[object Object]';
};
```
> ```js
this.isFunction = function (x) {
    return typeof x === 'function';
};
```

> And then could use like:

> ```js
if (AT._utils.isObject(x)) {
  // it's an object...
}
```

**Comment by nbrl** [comment]
A useful snippet for checking if something is *really* an object might be:

```js
Object.prototype.toString.call(thing) === '[object Object]' && thing !== null
```

`Object.prototype.toString.call(?):`

```
null       [object Object]
undefined  [object Object]
[]         [object Array]
{}         [object Object]
''         [object String]
```

**Comment by jabclab** [comment]
@nbrl good idea. Might also want to include a check for `undefined` there too? Would probably be worth consulting [`lodash`](https://github.com/lodash/lodash) for a completely safe way of doing it.

**Comment by nbrl** [comment]
@jabclab yep, not entirely sure why I wrote it that way. A seemingly cross-browser safe route is:

```js
thing && Object.prototype.toString.call(thing) === '[object Object]'
```

Though I agree, checking out `lodash` is a good call.
